syntax = "proto3";

package greet;

service Greeter {
  rpc sayHello (HelloRequest) returns (HelloReply);

  rpc parrotSaysHello(HelloRequest) returns (stream HelloReply);

  rpc chattyClientSaysHello(stream HelloRequest) returns (stream DelayedReply);

  rpc interactingHello (stream HelloRequest) returns (stream HelloReply);
}

message HelloRequest {
  string name = 1;
  string greeting = 2;
};

message HelloReply {
  string message = 1;
};

message DelayedReply {
  string message = 1;
  repeated HelloRequest request = 2;
}